astropy/cosmology/flrw/lambdacdm.py

```diff
@@ -95,8 +95,8 @@ def __post_init__(self):
         # about what is being done here.
         if self._Tcmb0.value == 0:
             inv_efunc_scalar = scalar_inv_efuncs.lcdm_inv_efunc_norel
-            inv_efunc_scalar_args = (self._Om0, self._Ode0, self._Ok0)
-            if self._Ok0 != 0:
+            inv_efunc_scalar_args = (self._Om0, self._Ode0, self.Ok0)
```

Comment:
Here, one place where things become less beautiful. Can one not handle all Omegas the same way?

Reply:
Some are `Parameter` objects, some are `cached_property`

Reply:
With the `Parameter` objects it's one less function call to access the private attributes.
Perhaps with https://peps.python.org/pep-0744/ and the Python 3.11 and 3.12 changes discussed therein this overhead can be minimized.